src/components/embedded/embedded-stores/EmbeddedMessagesStore.tsx

```diff
@@ -0,0 +1,210 @@
+import { isEventMessage } from '../../../helpers/event';
+import ApiSchema from '../../../api/ApiSchema';
+import { MessagesStoreURLState } from '../../../stores/messages/MessagesStore';
+import EmbeddedMessagesDataProviderStore from './EmbeddedMessagesDataProviderStore';
+import { MessagesSSEParams } from '../../../api/sse';
+
+function getDefaultMessagesFilter(): MessagesFilter {
+	const searchParams = new URLSearchParams(window.location.search);
+	const sessions: string[] = [];
+	const session = searchParams.get('session');
+
+	function defineSessions(): string[] {
+		if (session) sessions[0] = session;
+		return sessions;
+	}
+	return {
+		timestampFrom: null,
+		timestampTo: moment.utc().valueOf(),
+		streams: defineSessions(),
+	};
+}
+
+type MessagesStoreDefaultState = MessagesStoreURLState & {
+	targetMessage?: EventMessage;
+};
+
+export type MessagesStoreDefaultStateType = MessagesStoreDefaultState | string | null | undefined;
+
+export default class EmbeddedMessagesStore {
+	public dataStore = new EmbeddedMessagesDataProviderStore(this, this.api);
+
+	@observable
+	public hoveredMessage: EventMessage | null = null;
+
+	@observable
+	public selectedMessageId: String | null = null;
+
+	@observable
+	public scrolledIndex: Number | null = null;
+
+	@observable
+	public highlightedMessageId: String | null = null;
+
+	@observable
+	public detailedRawMessagesIds: Array<string> = [];
+
+	@observable
+	public beautifiedMessages: Array<string> = [];
+
+	@observable sseMessagesFilter: MessageFilterState | null = null;
+
+	@observable
+	public currentMessagesIndexesRange: ListRange = {
+		startIndex: 0,
+		endIndex: 0,
+	};
+
+	@observable filter: MessagesFilter = getDefaultMessagesFilter();
+
+	@observable
+	public showFilterChangeHint = false;
+
+	/*
+		  This is used for filter change hint. Represents either last clicked message
+		  or attached messages
+	  */
+	public hintMessages: EventMessage[] = [];
+
+	constructor(private api: ApiSchema, defaultState?: MessagesStoreDefaultStateType) {
+		this.init(defaultState);
+		reaction(() => this.selectedMessageId, this.onSelectedMessageIdChange);
+	}
+
+	private init = async (defaultState: MessagesStoreDefaultStateType) => {
+		if (!defaultState) {
+			return;
+		}
+		if (typeof defaultState === 'string') {
+			try {
+				const message = await this.api.messages.getMessage(defaultState);
+				this.onMessageSelect(message);
+			} catch (error) {
+				console.error(`Couldnt fetch target message ${defaultState}`);
+			}
+		} else {
+			const message = defaultState.targetMessage;
+			if (isEventMessage(message)) {
+				this.selectedMessageId = new String(message.messageId);
+				this.highlightedMessageId = message.messageId;
+			}
+		}
+		this.dataStore.loadMessages();
+	};
+
+	public get filterParams(): MessagesSSEParams {
+		const sseFilters = this.sseMessagesFilter;
+		const filtersToAdd: ('attachedEventIds' | 'type' | 'body' | 'bodyBinary')[] = [];
```

Comment:
Lets instead modify MessagesSSELoader so it accepts UrlSearchParams as queryParams, and we will just pass whatever is in url and add additional params, for now it will solve the problem